src/lib/utils/os_utils.h

```diff
@@ -148,6 +152,10 @@ void page_allow_access(void* page);
 */
 void page_named(void* page, size_t size);
 
+#if defined(BOTAN_TARGET_OS_HAS_THREADS)
+void set_thread_name(std::thread& thread, const char* const name);
```

Comment:
Is it required (by any OS) that this point to static storage? If so that should be documented here. Otherwise, `name` should be a `const std::string&` instead of a C style string.

(I would say a `std::string_view` but you can't convert that to a NULL terminated string)